functionalization fails to replace in-place operands with new tensors

*Note*: If you have a model or program that is not supported yet but should be, please use the program coverage template.

## 🐛 Bug

Functionalization fails to update some signatures in replacing in-place operands with new tensors.

<!-- If you have a code sample, error messages, stack traces, please provide it here as well -->

#### Code sample

`assert_close(e, e_)` in the snippet below fails with

```
Traceback (most recent call last):
  File "/home/mkozuki/ghq/github.com/Lightning-AI/lightning-thunder/issue_foobarbaz.py", line 33, in <module>
    main()
  File "/home/mkozuki/ghq/github.com/Lightning-AI/lightning-thunder/issue_foobarbaz.py", line 29, in main
    torch.testing.assert_close(e, e_)
  File "/home/mkozuki/ghq/github.com/crcrpar/pytorch/torch/testing/_comparison.py", line 1524, in assert_close
    raise error_metas[0].to_error(msg)
AssertionError: Tensor-likes are not close!

Mismatched elements: 4 / 4 (100.0%)
Greatest absolute difference: 0.4451824426651001 at index (0,) (up to 1e-05 allowed)
Greatest relative difference: 0.4193967282772064 at index (0,) (up to 1.3e-06 allowed)
```

```python
import torch
import thunder


def f(a: torch.Tensor, b: torch.Tensor) -> tuple[torch.Tensor, torch.Tensor]:
    c = torch.exp(a)
    d = torch.tanh(b)

    e = c.view(-1)
    e.add_(d.flatten())

    d.div_(a)
    return c, d, e / 2.0


def main():
    a, b = [torch.randn((2, 2), device="cuda", requires_grad=False) for _ in range(2)]
    a_, b_ = a.clone().detach(), b.clone().detach()
    jitted_f = thunder.jit(f)
    c, d, e = jitted_f(a, b)
    c_, d_, e_ = f(a_, b_)

    traces: list[thunder.core.trace.TraceCtx] = thunder.last_traces(jitted_f)
    print(traces[0])
    print(traces[2])

    torch.testing.assert_close(c, c_)
    torch.testing.assert_close(d, d_)
    torch.testing.assert_close(e, e_)


if __name__ == "__main__":
    main()
```

Traces before/after functionalization are as follows:
```python
def computation(a, b):
  # a: "cuda:0 f32[2, 2]"
  # b: "cuda:0 f32[2, 2]"

  # /home/mkozuki/ghq/github.com/Lightning-AI/lightning-thunder/issue_foobarbaz.py:6:       c = torch.exp(a)
  c = ltorch.exp(a)  # c: "cuda:0 f32[2, 2]"
    # c = prims.exp(a)  # c: "cuda:0 f32[2, 2]"

  # /home/mkozuki/ghq/github.com/Lightning-AI/lightning-thunder/issue_foobarbaz.py:7:       d = torch.tanh(b)
  d = ltorch.tanh(b)  # d: "cuda:0 f32[2, 2]"
    # d = prims.tanh(b)  # d: "cuda:0 f32[2, 2]"

  # /home/mkozuki/ghq/github.com/Lightning-AI/lightning-thunder/issue_foobarbaz.py:9:       e = c.view(-1)
  e = ltorch.view(c, -1)  # e: "cuda:0 f32[4]"
    # e = ltorch.reshape(c, (-1,))  # e: "cuda:0 f32[4]"
      # e = prims.reshape(c, (4,))  # e: "cuda:0 f32[4]"

  # /home/mkozuki/ghq/github.com/Lightning-AI/lightning-thunder/issue_foobarbaz.py:10:      e.add_(d.flatten())
  t3 = ltorch.flatten(d, 0, -1)  # t3: "cuda:0 f32[4]"
    # t3 = prims.reshape(d, (4,))  # t3: "cuda:0 f32[4]"
  t5 = ltorch.add_(e, t3, alpha=None)  # t5: "cuda:0 f32[4]"
    # t4 = ltorch.add(e, t3, alpha=None)  # t4: "cuda:0 f32[4]"
      # t4 = prims.add(e, t3)  # t4: "cuda:0 f32[4]"
    # t5 = prims.copy_(t4, e)  # t5: "cuda:0 f32[4]"

  # /home/mkozuki/ghq/github.com/Lightning-AI/lightning-thunder/issue_foobarbaz.py:12:      d.div_(a)
  t7 = ltorch.div_(d, a, rounding_mode=None)  # t7: "cuda:0 f32[2, 2]"
    # t6 = ltorch.div(d, a, rounding_mode=None, out=None)  # t6: "cuda:0 f32[2, 2]"
      # t6 = ltorch.true_divide(d, a)  # t6: "cuda:0 f32[2, 2]"
        # t6 = prims.div(d, a)  # t6: "cuda:0 f32[2, 2]"
    # t7 = prims.copy_(t6, d)  # t7: "cuda:0 f32[2, 2]"

  # /home/mkozuki/ghq/github.com/Lightning-AI/lightning-thunder/issue_foobarbaz.py:13:      return c, d, e / 2.0
  result = ltorch.true_divide(e, 2.0)  # result: "cuda:0 f32[4]"
    # result = prims.div(e, 2.0)  # result: "cuda:0 f32[4]"
  return (c, d, result)
```

```python
def computation(a, b):
  # a: "cuda:0 f32[2, 2]"
  # b: "cuda:0 f32[2, 2]"

  # /home/mkozuki/ghq/github.com/Lightning-AI/lightning-thunder/issue_foobarbaz.py:6:       c = torch.exp(a)
  c = ltorch.exp(a)  # c: "cuda:0 f32[2, 2]"
    # c = prims.exp(a)  # c: "cuda:0 f32[2, 2]"

  # /home/mkozuki/ghq/github.com/Lightning-AI/lightning-thunder/issue_foobarbaz.py:7:       d = torch.tanh(b)
  d = ltorch.tanh(b)  # d: "cuda:0 f32[2, 2]"
    # d = prims.tanh(b)  # d: "cuda:0 f32[2, 2]"

  # /home/mkozuki/ghq/github.com/Lightning-AI/lightning-thunder/issue_foobarbaz.py:9:       e = c.view(-1)
  e = ltorch.view(c, -1)  # e: "cuda:0 f32[4]"
    # e = ltorch.reshape(c, (-1,))  # e: "cuda:0 f32[4]"
      # e = prims.reshape(c, (4,))  # e: "cuda:0 f32[4]"

  # /home/mkozuki/ghq/github.com/Lightning-AI/lightning-thunder/issue_foobarbaz.py:10:      e.add_(d.flatten())
  t3 = ltorch.flatten(d, 0, -1)  # t3: "cuda:0 f32[4]"
    # t3 = prims.reshape(d, (4,))  # t3: "cuda:0 f32[4]"
  t4 = ltorch.add(e, t3, alpha=None)  # t4: "cuda:0 f32[4]"
    # t4 = ltorch.add(e, t3, alpha=None)  # t4: "cuda:0 f32[4]"
      # t4 = prims.add(e, t3)  # t4: "cuda:0 f32[4]"
  t6 = ltorch.div(d, a, rounding_mode=None, out=None)  # t6: "cuda:0 f32[2, 2]"
    # t6 = ltorch.div(d, a, rounding_mode=None, out=None)  # t6: "cuda:0 f32[2, 2]"
      # t6 = ltorch.true_divide(d, a)  # t6: "cuda:0 f32[2, 2]"
        # t6 = prims.div(d, a)  # t6: "cuda:0 f32[2, 2]"

  # /home/mkozuki/ghq/github.com/Lightning-AI/lightning-thunder/issue_foobarbaz.py:13:      return c, d, e / 2.0
  result = ltorch.true_divide(e, 2.0)  # result: "cuda:0 f32[4]"
    # result = prims.div(e, 2.0)  # result: "cuda:0 f32[4]"
  t9 = prims.reshape(t4, (2, 2))  # t9: "cuda:0 f32[2, 2]"

  # /home/mkozuki/ghq/github.com/Lightning-AI/lightning-thunder/issue_foobarbaz.py:13:      return c, d, e / 2.0
  return (t9, t6, result)
```

`result = ltorch.true_divide(e, 2.0)` should be `result = ltorch.true_divide(t4, 2.0)`

--- a/thunder/core/transform_common.py
+++ b/thunder/core/transform_common.py
@@ -505,7 +505,7 @@ def functionalize_inplace_ops(
                     reshape_bsym = prims.reshape.bind(view, unvariableify(var_orig).shape, output=view_of_orig_shape)
                     cur_orig_to_view_swap_map[var_orig] = view_of_orig_shape
                     bsyms.append(reshape_bsym)
-        new_bsym = bsym.from_bsym_swap_proxies(cur_orig_to_view_swap_map, skip_output=True)
+        new_bsym = new_bsym.from_bsym_swap_proxies(cur_orig_to_view_swap_map, skip_output=True)
 
         # in-place functionalizable ops has `prims.copy_` as the last subsymbol.
         if not is_functionalizable(new_bsym):

--- a/thunder/tests/test_inplace_functionalization.py
+++ b/thunder/tests/test_inplace_functionalization.py
@@ -210,6 +210,26 @@ def test_inplace_to_views(executor, device, _):
 
     torch.testing.assert_close((d, e), (d_, e_))
 
+    def h(a: torch.Tensor, b: torch.Tensor) -> tuple[torch.Tensor, torch.Tensor]:
+        c = torch.exp(a)
+        d = torch.tanh(b)
+
+        e = c.view(-1)
+        e.add_(d.flatten())
+
+        d.div_(a)
+        return c, d, e / 2.0
+
+    a, b = (make_tensor((2, 2), device=device, dtype=torch.float32) for _ in range(2))
+    a_, b_ = a.clone().detach(), b.clone().detach()
+
+    jittd_h = thunder.jit(h, executors=executor.executors_list())
+
+    c, d, e = jittd_h(a, b)
+    c_, d_, e_ = h(a_, b_)
+
+    torch.testing.assert_close((c, d, e), (c_, d_, e_))
+
 
 @instantiate(
     dtypes=NOTHING,

